test/provider_fixed_memory.cpp

```diff
@@ -334,11 +339,55 @@ TEST_F(test, create_with_zero_size) {
     ASSERT_EQ(wrong_params, nullptr);
 }
 
-TEST_P(FixedProviderTest, alloc_size_exceeds_buffer) {
-    size_t size = memory_size + page_size;
-    test_alloc_failure(size, 0, UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY, 0);
+TEST_F(test, params_several_set_memory) {
+    constexpr size_t memory_size1 = 100, memory_size2 = 1, memory_size3 = 250;
+    char memory_buffer1[memory_size1];
+    char memory_buffer2[memory_size2];
+    char memory_buffer3[memory_size3];
+
+    umf_fixed_memory_provider_params_handle_t params = nullptr;
```

Comment:
this scenario tests almost nothing. Please test following scenario:
* create fixed provider params with ptr/size set to 0
* then create buffer A and call SetMemory with &A
* then create provider, make allocation and check if the allocation is valid and is a part of A,
* then create buffer B and repeat above steps for B with the same params struct

Reply:
can't set size/ptr to 0, but otherwise done